### PR TITLE
Dependency Graph Builder Mojo Skeleton

### PR DESCRIPTION
**Pull request by ian-lavallee**

### CODE REVIEW

File: verboseTree-maven-plugin/pom.xml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.apache.maven.plugin-testing</groupId>
+      <artifactId>maven-plugin-testing-harness</artifactId>
+      <version>3.3.0</version>
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>1.10.19</version>
```
Comment (elharo) [inline verboseTree-maven-plugin/pom.xml:63]: 2.28.2

File: verboseTree-maven-plugin/pom.xml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>2.28.2</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-compat</artifactId>
+      <version>3.6.3</version>
```
Comment (elharo) [inline verboseTree-maven-plugin/pom.xml:69]: scope test

File: verboseTree-maven-plugin/pom.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  <name>VerboseTreePlugin</name>
+  <description>a plugin to output the verbose dependency tree</description>
+
+  <dependencies>
+    <!-- dependencies needed for a maven plugin -->
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-plugin-api</artifactId>
+      <version>3.6.3</version>
```
Comment (elharo) [inline verboseTree-maven-plugin/pom.xml:20]: make 3.6.3 a maven-version project variable

File: verboseTree-maven-plugin/pom.xml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      <groupId>org.apache.maven.plugin-tools</groupId>
+      <artifactId>maven-plugin-annotations</artifactId>
+      <version>3.6.0</version>
+      <scope>provided</scope>
+    </dependency>
+    <!-- Needed for BuildingRequest Mojo -->
+    <dependency>
+      <groupId>org.apache.maven.resolver</groupId>
+      <artifactId>maven-resolver-util</artifactId>
+      <version>1.4.2</version>
```
Comment (elharo) [inline verboseTree-maven-plugin/pom.xml:39]: make 1.4.2 a maven-resolver-version project variable

File: verboseTree-maven-plugin/pom.xml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.13</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.maven.plugin-testing</groupId>
+      <artifactId>maven-plugin-testing-harness</artifactId>
+      <version>3.3.0</version>
```
Comment (elharo) [inline verboseTree-maven-plugin/pom.xml:58]: scope test

File: verboseTree-maven-plugin/src/test/java/org/apache/maven/dependency/graph/DependencyGraphBuilderTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.junit.Test;
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+public class DependencyGraphBuilderTest extends AbstractMojoTestCase
+{
+    DependencyGraphBuilder mojo;
```
Comment (elharo) [inline verboseTree-maven-plugin/src/test/java/org/apache/maven/dependency/graph/DependencyGraphBuilderTest.java:34]: private

File: verboseTree-maven-plugin/pom.xml [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+      <scope>test</scope>
+    </dependency>
+    <dependency>
```
Comment (elharo) [inline verboseTree-maven-plugin/pom.xml:67]: You don't seem to be using this dependency anywhere.

Comment (ian-lavallee) [reply verboseTree-maven-plugin/pom.xml:67]: This is where AbstractMojoTestCase comes from

**Pull request merged**